ucsb-cs156-f23/proj-happycows-f23-5pm-3
# ky - updated chat feature with new ui designs

## Overview
In this PR, the UI for the chat feature on the Play page has been updated. There is a new set of icons used to denote when the chat is open and when it is closed. The designs seek to capture the user's attention through vivid colors and visually interesting illustrative elements. The chat toggle functionality works as intended. The toggle chat button hovers above the rest of the content and is fixed even as the user scrolls. 
Furthermore, a new ChatFeature component which combines the ChatPanel and the chat toggle has been created, facilitating more generalized usability.
All new image assets were designed in Figma.

## Storybook link

https://ucsb-cs156-f23.github.io/proj-happycows-f23-5pm-3/prs/26/storybook/?path=/story/pages-playpage--default

## Screenshots
<img width="1434" alt="Chat closed" src="https://github.com/ucsb-cs156-f23/proj-happycows-f23-5pm-3/assets/26288908/0fd9f878-1905-4229-a8b1-38b4dc3ee081">
Fig 1. The UI when the chat is closed. Note the drop shadow to enhance the visual presence of the button and the speech bubble to indicate what action the button performs.
<br /><br />
<img width="1435" alt="Chat opened" src="https://github.com/ucsb-cs156-f23/proj-happycows-f23-5pm-3/assets/26288908/c8608ea6-f024-405c-8959-5e389a9e0902">
Fig 2. The UI when the chat is opened. Notice the chat panel displayed above the button.

## Feedback Request
Please give suggestions on the following:
- Thoughts on the designs and user flow? Is it intuitive? Are users able to identify the functionality of the feature?

## Validation

1. Download this branch.
2. Run the project.
3. Log in.
4. Visit a Commons (if no commons are present, create one and join it. Then visit it).
5. Click that chat button in the lower right corner to toggle the visibility of the chat.
6. Optional: send messages in the chat to verify that the feature is functioning as intended.

## Tests


## Linked Issues

Closes #7
